scripts/all-dependencies.sh

```diff
@@ -56,8 +56,8 @@ dep boringssl libcrypto.a libssl.a libdecrepit.a
 dep cares libcares.a
 dep libarchive libarchive.a
 dep lolhtml liblolhtml.a
-dep mimalloc-debug libmimalloc-debug.a
-dep mimalloc libmimalloc.a
+dep mimalloc-debug libmimalloc-debug.a libmimalloc-debug.o
+dep mimalloc libmimalloc.a libmimalloc.o
```

Comment:
What is the rationale for using `.o` here?

Reply:
It supposedly allows mimalloc to override `malloc` easier, but I'm not sure. this logic is ported from the old Makefile.

```make
_MIMALLOC_FILE = libmimalloc.o
_MIMALLOC_INPUT_PATH = CMakeFiles/mimalloc-obj.dir/src/static.c.o
```
```make
ifeq ($(OS_NAME),darwin)
  _MIMALLOC_OBJECT_FILE = 0
  _MIMALLOC_FILE = libmimalloc.a
  _MIMALLOC_INPUT_PATH = libmimalloc.a
  _MIMALLOC_LINK = -lmimalloc
endif
```

Not sure if this claim is still relevant, or any performance differences exist, but it is easier to keep the old system going.